### PR TITLE
docs: add documentation for missing IDPs

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/33/

File: docs/configuration.md [hunk 1]
```diff
@@ -226,6 +226,115 @@ will be used.
...
+    email: mail
+    display-name: displayName
+  group-query-filter: (&(objectClass=groupOfNames)(member={{.User}}))
+```
+
+The LDAP identity provider allows a user to login using an LDAP server.
+Candid will prompt for a username and password and attempt to use those
+to authenticate with the LDAP server.
+
+`name` is the name to use for the LDAP IDP instance, it is possible
```
Comment (rogpeppe) [inline docs/configuration.md]: s/, it/. It/

File: docs/configuration.md [hunk 2]
```diff
@@ -226,6 +226,115 @@ will be used.
...
+to configure more than one LDAP IDP on a given candid server and this
+allows them to be identified, name will be used in the login URL.
```
Comment (rogpeppe) [inline docs/configuration.md]: s/, name will /. The name will/

File: docs/configuration.md [hunk 3]
```diff
@@ -226,6 +226,115 @@ will be used.
...
+redirect URLs should include `$CANDID_URL/login/azure/callback`.
+
+### Google OpenID Connect
+```yaml
+- type: google
+  client-id: 483156874216-rh0j89ltslhuqirk7deh70d3mp49kdvq.apps.googleusercontent.com
+  client-secret: 8aENrwCL/+PU87ROkXwMB+09xe0=
+```
+
+The Google identity provider uses OpenID connect to log in using Google
```
Comment (rogpeppe) [inline docs/configuration.md]: s/connect/Connect/ ?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/34/

File: docs/configuration.md [hunk 4]
```diff
@@ -226,6 +226,115 @@ will be used.
...
+`user-query-attrs` contains the attributes candid uses when searching
+for authenticating users.  When authenticating a user candid will
+perform a search like `($id=$username)` where the value of `$id` is
+specified in the `id` parameter and $username is the value entered by
+the authenticating user. `email` and `display-name` are used to populate
+the created identity.
+
+`group-query-filter` contains the filter candid uses when finding group
+memberships for a user.  The filter is specified as a template (see
+https://golang.org/pkg/text/template) where the value of .User will be
```
Comment (albertodonato) [inline docs/configuration.md]: maybe backtick-quote `.User` too ?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/35/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/36/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**